pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -140,9 +140,19 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(builder *ListenerBui
 				// we will support more cases)
 				newFilterChains = configgen.buildGatewayHTTP3FilterChains(builder, serversForPort, mergedGateway, proxyConfig, opts)
 			}
-			var mutable *MutableListener
+
+			for cnum := range newFilterChains {
+				if util.IsIstioVersionGE117(builder.node.IstioVersion) {
+					newFilterChains[cnum].TCP = append(newFilterChains[cnum].TCP, xdsfilters.IstioNetworkAuthenticationFilter)
+				}
+				if newFilterChains[cnum].ListenerProtocol == istionetworking.ListenerProtocolTCP {
```

Comment:
So in 1.16 this part (authzBuilder and authzCustomBuilder) was below. Was that also broken and just not noticed?

This looks right, just trying to make sure I understand since this is very sensitive code

Reply:
Yes, I've just checked in the form of a unit test that `"envoy.filters.network.rbac"` filter gets inserted twice on both `Istio 1.16` and `Istio 1.17`.

I've used the same test case as in this PR but with the reverse order of servers in the `Gateway` object: `TLS` first, `HTTPS` second.

I assume it wasn't a critical problem for `"envoy.filters.network.rbac"` filter since it's probably idempotent.